### PR TITLE
Bump React deps to v19 stable release

### PR DESCRIPTION
**Pull request by jerelmiller**
Updates our dev dependency and allowed peer dependency range for React 19.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -74,8 +74,8 @@
   "peerDependencies": {
     "graphql": "^15.0.0 || ^16.0.0",
     "graphql-ws": "^5.5.5",
-    "react": "^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc <19.0.0",
-    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc <19.0.0",
+    "react": "^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc",
+    "react-dom": "^16.8.0 || ^17.0.0 || ^18.0.0 || >=19.0.0-rc",
```
Comment (jerelmiller) [inline package.json:78]: Did we want to continue supporting the `rc` release for a time? If not, I can change this to `^19.0.0`.

Comment (phryneas) [reply package.json:78]: Yeah, no reason not to.

Comment (phryneas) [reply package.json:78]: Although we surely will get PRs to change it to 19.
Just calling it 🤣

**Review by phryneas — APPROVED** [review]
Once CI passes: approved!

File: .circleci/config.yml [hunk 2]
```diff
@@ -181,7 +181,7 @@ workflows:
                 - "@types/react@16.8 @types/react-dom@16.8"
                 - "@types/react@17 @types/react-dom@17"
                 - "@types/react@18 @types/react-dom@18"
-                - "@types/react@npm:types-react@19.0.0-rc.1 @types/react-dom@npm:types-react-dom@19.0.0-rc.1"
+                - "@types/react@npm:types-react@19.0.0 @types/react-dom@npm:types-react-dom@19.0.0"
```
Comment (phryneas) [inline .circleci/config.yml:184]:
```suggestion
                - "@types/react@npm:types-react@19.0.0 @types/react-dom@19.0.0"
```

File: package.json [hunk 3]
```diff
@@ -170,10 +170,10 @@
     "prettier": "3.1.1",
     "react": "18.3.1",
     "react-17": "npm:react@^17",
-    "react-19": "npm:react@19.0.0-rc.1",
+    "react-19": "npm:react@19.0.0",
     "react-dom": "18.3.1",
     "react-dom-17": "npm:react-dom@^17",
-    "react-dom-19": "npm:react-dom@19.0.0-rc.1",
+    "react-dom-19": "npm:react-dom@19.0.0",
```
Comment (phryneas) [inline package.json:176]: Do we also have types we need to bump?

Comment (jerelmiller) [reply package.json:176]: [49b263e](https://github.com/apollographql/apollo-client/pull/12175/commits/49b263ef75767453b17572113795733be378a3e8)

Comment (jerelmiller) [reply package.json:176]: Removing for now and will split that out to a separate PR

**Pull request merged**